modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +47,402 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
+</root>
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+ns html http://www.w3.org/TR/html4/
+ns furniture https://www.w3schools.com/furniture
+---
+root: {
+     a: payload.root.html#table,
+     b: payload.root.furniture#table
+ }    
+----
+
+==== Output
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+  <b>
+    <f:name xmlns:f="https://www.w3schools.com/furniture">African Coffee Table</f:name>
+    <f:width xmlns:f="https://www.w3schools.com/furniture">80</f:width>
+    <f:length xmlns:f="https://www.w3schools.com/furniture">120</f:length>
+  </b>
+</root>
+----
+
+=== Example
+
+Shows how to select an xml with local name, matching an element with namespace
+
+==== Input
+
+[source,xml, linenums]
+----
+<root>
+
+<h:table xmlns:h="http://www.w3.org/TR/html4/">
+  <h:tr>
+    <h:td>Apples</h:td>
+    <h:td>Bananas</h:td>
+  </h:tr>
+</h:table>
+
+
+</root>
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+ns html http://www.w3.org/TR/html4/
+ns furniture https://www.w3schools.com/furniture
+---
+root: {
+     a: payload.root.table
+ }    
+----
+
+==== Output
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>  
+</root>
+----
+
+== Attribute Selector
+
+The attribute selector returns the first Attribute value that matches the selected name expression. If no key was matched it return `null`
+
+=== Example
+
+Shows how to select an xml attribute
+
+==== Input
+
+[source,xml, linenums]
+----
+<user name="Weave"/>
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.user.@name   
+----
+
+==== Output
+
+[source,json,linenums]
+----
+"Weave"
+----
+
+
+== Multi Value Selector
+
+Returns an Array of all the matched keys value. If no key was matched it return `null`
+
+=== Example
+
+Shows how to select multiple values
+
+==== Input
+
+[source,xml, linenums]
+----
+<users>
+    <user>Weave</user>
+    <user>BAT</user>
+    <user>TF</user>
+</users>
+----
+
+==== Source
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.users.*user
+----
+
+==== Output
+
+[source,json,linenums]
+----
+[
+  "Weave",
+  "BAT",
+  "TF"
+]
+----
+
+
+== Descendant Selector
+
+Returns the list of all children. This selector can be chained with any other selector directly without requiering any `.`. For example
+`payload..` returns all the child values and the value of its children and recursively. But it can also be chained with `payload..name` where it will select the name to all its descendants, or `payload..*name` where it will select all the names each descendant may have.
```

Comment:
Probably need a different example, as `payload..name` vs. `payload..*name` return the same output when given the input below. Could we make the point and show different results using something like `payload..user` and `payload..*user` on a different input, one that adds a second set of `user` elements, for example `<users>
  <user>
    <name>Weave</name>
    <user>
      <name>BAT</name>
      <user>
        <name>BDD</name>
      </user>
    </user>
  </user>
  <user>
    <name>Weave2</name>
    <user>
      <name>BAT2</name>
      <user>
        <name>BDD2</name>
      </user>
    </user>
  </user>
</users>`

Reply:
In this case, `payload..user`  returns: 
```
[
  {
    "name": "Weave",
    "user": {
      "name": "BAT",
      "user": {
        "name": "BDD"
      }
    }
  },
  {
    "name": "BAT",
    "user": {
      "name": "BDD"
    }
  },
  {
    "name": "BDD"
  },
  {
    "name": "BAT2",
    "user": {
      "name": "BDD2"
    }
  },
  {
    "name": "BDD2"
  }
]
```

Reply:
And `payload..*user` returns this (which includes the `user` element with the "Weave2" value):  
```
[
  {
    "name": "Weave",
    "user": {
      "name": "BAT",
      "user": {
        "name": "BDD"
      }
    }
  },
  {
    "name": "Weave2",
    "user": {
      "name": "BAT2",
      "user": {
        "name": "BDD2"
      }
    }
  },
  {
    "name": "BAT",
    "user": {
      "name": "BDD"
    }
  },
  {
    "name": "BDD"
  },
  {
    "name": "BAT2",
    "user": {
      "name": "BDD2"
    }
  },
  {
    "name": "BDD2"
  }
]
```